modules/file-extension-common/src/main/java/org/mule/runtime/module/extension/file/api/FilePredicateBuilder.java

```diff
@@ -192,12 +195,14 @@ public T setSymbolicLink(Boolean symbolicLink)
 
     public T setMinSize(Long minSize)
     {
+        checkArgument(minSize >= 0, format(SIZE_MUST_BE_POSITIVE_MESSAGE, "minSize", minSize));
```

Comment:
this is the wrong place to do this. You should do this on the build() method, since the setter is not invoked when the object is create through the SDK.